core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -241,10 +243,17 @@ public void messageRead(final InputStream message) {
         messageReceived = true;
 
         listener.onMessage(call.method.parseRequest(message));
+      } catch (Throwable e) {
+        t = e;
       } finally {
         try {
           message.close();
         } catch (IOException e) {
+          if (t != null) {
+            // TODO(carl-mastrangelo): Maybe log e here.
+            Throwables.propagateIfPossible(t);
```

Comment:
propagateIfPossible is the official replacement for propagate, it is not deprecated.  propagate was being removed /because/ it was such a simple wrapper around propagateIfPossible